pilot/pkg/security/authz/model/principal.go

```diff
@@ -72,7 +72,7 @@ func (principal *Principal) ValidateForTCP(forTCP bool) error {
 
 	for _, p := range principal.Properties {
 		for key := range p {
-			if strings.HasPrefix(key, "request.auth.") || key == attrSrcUser {
```

Comment:
What is the reason behind this? Do we document what properties are *not* for TCP?

Reply:
This is to include `request.headers`, the attributes are documented here: https://istio.io/docs/reference/config/security/conditions/

Reply:
some comment may be useful. The loop goes through `principal.Properties` which I often associate with the authn attributes.

Reply:
Added comment.